mempool/v0/reactor.go

```diff
@@ -181,42 +201,224 @@ func (memR *Reactor) ReceiveEnvelope(e p2p.Envelope) {
+		if !memR.IsRunning() || !peer.IsRunning() {
+			return
+		}
+		// This happens because the CElement we were looking at got garbage
+		// collected (removed). That is, .NextWait() returned nil. Go ahead and
+		// start from the beginning.
+		if next == nil {
+			select {
+			case <-memR.sidecar.TxsWaitChan(): // Wait until a tx is available in sidecar
+				// if a tx is available on sidecar, if fire is set too, then fire
+				if next = memR.sidecar.TxsFront(); next == nil {
+					continue
+				}
+			case <-peer.Quit():
+				return
+			case <-memR.Quit():
+				return
+			}
+		}
+		// Make sure the peer is up to date.
+		peerState, ok := peer.Get(types.PeerStateKey).(PeerState)
+		if !ok {
+			// Peer does not have a state yet. We set it in the consensus reactor, but
+			// when we add peer in Switch, the order we call reactors#AddPeer is
+			// different every time due to us using a map. Sometimes other reactors
+			// will be initialized before the consensus reactor. We should wait a few
+			// milliseconds and retry.
+			time.Sleep(mempool.PeerCatchupSleepIntervalMS * time.Millisecond)
+			continue
+		}
+		if scTx, okConv := next.Value.(*mempool.SidecarTx); okConv && isSidecarPeer {
+			memR.Logger.Debug(
+				"broadcasting a sidecarTx to peer",
+				"peer", peerID,
+				"tx", scTx.Tx.Hash(),
+			)
+			if peerState.GetHeight() < scTx.DesiredHeight-1 {
+				time.Sleep(mempool.PeerCatchupSleepIntervalMS * time.Millisecond)
+				continue
+			}
+			if _, ok := scTx.Senders.Load(peerID); !ok {
+				// try sending over using envelop protocol over new channel first
+				msg := &protomem.MEVTxs{
+					Txs:           [][]byte{scTx.Tx},
+					DesiredHeight: scTx.DesiredHeight,
+					BundleId:      scTx.BundleID,
+					BundleOrder:   scTx.BundleOrder,
+					BundleSize:    scTx.BundleSize,
+					GasWanted:     scTx.GasWanted,
+				}
+				success := p2p.SendEnvelopeShim(peer, p2p.Envelope{ //nolint: staticcheck
+					ChannelID: mempool.SidecarChannel,
+					Message:   msg,
+				}, memR.Logger)
+				if !success {
+					// if sending over envelop protocol fails, try sending over legacy protocol
+					msg := &protomem.MEVMessage{
+						Sum: &protomem.MEVMessage_Txs{
+							Txs: &protomem.Txs{Txs: [][]byte{scTx.Tx}},
+						},
+						DesiredHeight: scTx.DesiredHeight,
+						BundleId:      scTx.BundleID,
+						BundleOrder:   scTx.BundleOrder,
+						BundleSize:    scTx.BundleSize,
+						GasWanted:     scTx.GasWanted,
+					}
+
+					success := p2p.SendEnvelopeShim(peer, p2p.Envelope{
+						ChannelID: mempool.SidecarLegacyChannel,
+						Message:   msg,
+					}, memR.Logger)
+					if !success {
+						time.Sleep(mempool.PeerCatchupSleepIntervalMS * time.Millisecond)
+						continue
+					}
+				}
+			} else {
+				memR.Logger.Info(
+					"broadcasting sidecarTx to peer failed",
+					"peer", peerID,
+					"was considered sidecarPeer", isSidecarPeer,
+					"was converted to sidecarTx", okConv,
+					"tx", scTx.Tx.Hash(),
+				)
+			}
```

Comment:
This log message could be misleading, because here we are not having issues with  broadcasting sidecarTx failure but the broadcasting is simply optimized and the peer is skipped if being the one that sent the transactions to the node.
My suggestion is to remove this logging since it is expected behavior or least modify the info message.

There is a same else branch in v1/reactor.go.

Reply:
Listed in https://github.com/informalsystems/audit-skip/issues/6

Reply:
Removed in https://github.com/skip-mev/mev-tendermint/pull/131